README.md

```diff
@@ -34,7 +42,110 @@ Or install it yourself as:
 
     $ gem install transproc
 
-## Usage
+## Basics
+
+The simple transformations are defined as easy as
+
+```ruby
+increament = Transproc::Function.new(-> (data) { data + 1 })
+increament[1] # => 2
+```
+
+It's easy to compose transformations
+
+```ruby
+to_string = Transproc::Function.new(:to_s.to_proc)
+(increament >> to_string)[1] => '2'
+```
+
+It's easy to use extra arguments to transformation
+
+```ruby
+append = Transproc::Function.new(-> (value, suffix) { value + suffix })
+append_bar = append.with('_bar')
+append_bar['foo'] # => foo_bar
+```
+
+Or even accept other transformation as argument
+
+```ruby
+map_array = Transproc::Function.new(-> (array, fn) { array.map(&fn) })
+map_array.with(to_string).call([1, 2, 3]) # => ['1', '2', '3']
+```
+
+To improve this low-level definition, you can use class methods
+with `Transproc::Registry`
+
+```ruby
+M = Module.new do
+  extend Transproc::Registry
+
+  def self.to_string(value)
+    value.to_s
+  end
+
+  def self.map_array(array, fn)
+    array.map(&fn)
+  end
+end
+M[:map_array, M[:to_string]].([1, 2, 3]) # => ['1', '2', '3']
+```
+
+### Built-in transformations
+
+`transproc` comes with a lot of built-in functions. They come in form of module
```

Comment:
Should be `They come in the form of modules with class methods, which you can import into a registry.